bson/json_util.py

```diff
@@ -497,7 +497,11 @@ def loads(s: Union[str, bytes, bytearray], *args: Any, **kwargs: Any) -> Any:
        Accepts optional parameter `json_options`. See :class:`JSONOptions`.
     """
     json_options = kwargs.pop("json_options", DEFAULT_JSON_OPTIONS)
-    kwargs["object_pairs_hook"] = lambda pairs: object_pairs_hook(pairs, json_options)
+    # Execution time optimization if json_options.document_class is dict
+    if json_options.document_class is dict:
```

Comment:
```suggestion
    if isinstance(json_options.document_class, dict):
```

Using `isinstance` supports inheritance, `is dict` only works for a literal `dict` object.

Reply:
This code will go away with my suggestion to always use object_hook above.

Reply:
I suppose what was meant here is issubclass(json_options.document_class, dict) because
```
>>> isinstance(dict, dict)
False
```
However, if you simply use issubclass, loads will return an object of the incorrect type, specifically always just dict.
Similar problem will occur if object_pairs_hook is completely removed without additional modifications. But I think it's worth trying to reduce the number of calls to the constructor of the dict-like class, which could optimize the execution time.

Reply:
Additionally, I've noticed that the behavior I described isn't being tested. 
https://github.com/mongodb/mongo-python-driver/blob/master/test/test_json_util.py#L565 This test looks as though it's testing, but in reality, it doesn't check the type, and such a test
 self.assertEqual(
            SON([("foo", "bar"), ("b", 1)]),
            {"foo": "bar", "b": 1}
        )
 will run without errors.

Reply:
Good catch. We need to add assertions that top-level and embedded objects are all SON with document_class=SON.

Reply:
Update: I'm added this test here: https://github.com/mongodb/mongo-python-driver/pull/1509